sklearn/preprocessing/tests/test_discretization.py

```diff
@@ -172,5 +177,43 @@ def test_numeric_stability():
     # Test up to discretizing nano units
     for i in range(1, 9):
         X = X_init / 10**i
-        Xt = KBinsDiscretizer(n_bins=2).fit_transform(X)
+        Xt = KBinsDiscretizer(n_bins=2, encode='ordinal').fit_transform(X)
         assert_array_equal(Xt_expected, Xt)
+
+
+def test_invalid_encode_option():
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3], encode='invalid-encode')
+    assert_raises(ValueError, est.fit, X)
+
+
+def test_encode_options():
+    # test valid encode options through comparison
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3],
+                           encode='ordinal').fit(X)
+    Xt_1 = est.transform(X)
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3],
+                           encode='onehot-dense').fit(X)
+    Xt_2 = est.transform(X)
+    assert not sp.issparse(Xt_2)
+    assert_array_equal(OneHotEncoder(n_values=[2, 3, 3, 3], sparse=False)
+                       .fit_transform(Xt_1), Xt_2)
+    assert_raises(ValueError, est.inverse_transform, Xt_2)
```

Comment:
Could you match the error message (or part of it) using `assert_raises_regex`

In addition I would group all the error inside a side test.